src/helpers/helpers.time.js

```diff
@@ -55,15 +55,28 @@ module.exports = function(Chart) {
 		var ticks = [];
 		if (options.maxTicks) {
 			var stepSize = options.stepSize;
-			ticks.push(options.min !== undefined ? options.min : niceRange.min);
-			var cur = moment(niceRange.min);
-			while (cur.add(stepSize, options.unit).valueOf() < niceRange.max) {
-				ticks.push(cur.valueOf());
+			var startTick = options.min !== undefined ? options.min : niceRange.min;
+			var majorUnitStart = startTick;
+			var majorUnit = options.majorUnit;
+			if (majorUnit) {
+				majorUnitStart = moment(startTick).add(1, majorUnit).startOf(majorUnit);
+			}
+			var startRange = majorUnitStart.valueOf() - startTick;
+			var stepValue = interval[options.unit].size * stepSize;
+			var startFraction = startRange % stepValue;
+			var alignedTick = startTick;
+			if (startFraction && majorUnit && !options.timeOpts.round && !options.timeOpts.isoWeekday) {
+				alignedTick += startFraction - stepValue;
```

Comment:
Is `alignedTick += startFraction - stepValue` equal to `alignedTick -= startFraction` ?  If so, that might be a more straightforward way of writing it